util/http/client/client_test.go

```diff
@@ -25,7 +27,40 @@ import (
 	. "github.com/smartystreets/goconvey/convey"
 )
 
-func TestHTTPClient_Fetch(t *testing.T) {
+func TestClient(t *testing.T) {
+	Convey("User-Agent should match Git version", t, func() {
+		var gotUserAgent string
+
+		ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			td, err := json.Marshal(map[string]string{
+				"foo": "bar",
+			})
+			if err != nil {
+				panic(err)
+			}
+
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusOK)
+			w.Write(td)
+
+			gotUserAgent = r.UserAgent()
+		}))
+		defer ts.Close()
+
+		u, err := url.Parse(ts.URL)
+		if err != nil {
+			panic(err)
+		}
+
+		if err := Fetch(http.DefaultClient, *u, nil); err != nil {
+			panic(err)
+		}
+
+		So(gotUserAgent, ShouldContainSubstring, "com.mesosphere.dcos-metrics/")
```

Comment:
@malnick -- I'd like to do something similar to what you did with the linker flags for setting the `userAgent` string, but that appears to only happen on a `go build` and not during `go test`. I can easily modify the ldflags in the build script, but I'm not sure how I'd actually test it. Do you have any ideas here?